contracts/exchange/contracts/src/MixinTransactions.sol

```diff
@@ -96,6 +140,16 @@ contract MixinTransactions is
             ));
         }
 
+        // Validate that transaction is executed with the correct gasPrice
+        uint256 requiredGasPrice = transaction.gasPrice;
+        if (tx.gasprice != requiredGasPrice) {
+            LibRichErrors.rrevert(LibExchangeRichErrors.TransactionGasPriceError(
+                transactionHash,
+                tx.gasprice,
+                requiredGasPrice
+            ));
+        }
+
         // Prevent reentrancy
         if (currentContextAddress != address(0)) {
```

Comment:
I think that there is a way for us to support nested `executeTransaction` calls. If we kept a stack variable called `startingContextAddress`, we could then replace our erasure of the `currentContextAddress` with the `startingContextAddress`. In this way, we'd get the best of both worlds by allowing users to submit metatransactions that submit other metatransactions (relayers could possibly make use of this), but they are always executed in the right context. Thoughts?